src/Common/src/Interop/Unix/libssl/Interop.libssl_types.cs

```diff
@@ -61,12 +62,43 @@ internal static class SslSizes
             internal const int TRAILER_SIZE = 68;
         }
 
-        internal static class SslErrorCode
+        internal static class SslMethods
         {
-            internal const int SSL_ERROR_SSL = 1;
-            internal const int SSL_ERROR_WANT_READ = 2;
-            internal const int SSL_ERROR_SYSCALL = 5;
-            internal const int SSL_ERROR_ZERO_RETURN = 6;
+            internal static readonly IntPtr TLSv1_method;
+            internal static readonly IntPtr TLSv1_1_method;
+            internal static readonly IntPtr TLSv1_2_method;
+            internal static readonly IntPtr SSLv3_method;
+            internal static readonly IntPtr SSLv23_method;
+
+            private static IntPtr checkMethodPtr(IntPtr methodPtr)
```

Comment:
Without line numbers in stack traces right now on Unix, we won't know which of the calls to checkMethodPtr is causing the assert.  Since the whole point of this method is the Debug.Fail, I'd suggest adding a [CallerLineNumber] argument to the method, and using that line number in the call to Debug.Fail, so that we know exactly which one fails if it does.